lib/src/charlatan_http_client_adapter.dart

```diff
@@ -44,7 +44,7 @@ class CharlatanHttpClientAdapter implements HttpClientAdapter {
 
 Unable to find matching fake http response definition for:
 
-${method.toUpperCase()} $path
+${method.toUpperCase()} $path ${options.data}
```

Comment:
> We could add a hook on Charlatan to allow you to customize how the request prints out for this debugging use case with a solid default.

I like this idea as it was something I had been thinking about for the actual "matching" logic. I was struggling to suss out what Charlatan was doing under the hood and breakpoints were helpful but didn't _fully_ solve my struggles so I pulled down Charlatan locally and added some logs like:

```console
Matching $requestInfo against $matcherDescription; did not match
Matching $requestInfo against $otherMatcherDescription; did match
...
```

That along with the actual "couldn't find matches, here's what's configured:" message could be value adds, but I'd like to iterate on that in a follow-up PR because the real need from this PR is the exports of the `typedef`s. So I'll back these changes out and maybe open an issue for this stuff.